src/mobu/storage/github.py

```diff
@@ -0,0 +1,339 @@
+        client: GitHubAPI,
+        repo_owner: str,
+        repo_name: str,
+        ref: str,
+    ) -> None:
+        self.client = client
+        self.repo_owner = repo_owner
+        self.repo_name = repo_name
+        self.ref = ref
+        self.commit_url = f"https://github.com/{self.repo_owner}/{self.repo_name}/commit/{self.ref}"
+        self._api_path = f"/repos/{repo_owner}/{repo_name}"
+
+    @classmethod
+    async def create(
+        cls,
+        factory: GitHubAppClientFactory,
+        installation_id: int,
+        repo_owner: str,
+        repo_name: str,
+        ref: str,
+    ) -> Self:
+        """Create an auth'd GitHub client and construct an instance.
+
+        Parameters
+        ----------
+        factory
+            A GitHub client factory with credentials.
+        installation_id
+            The ID of an installed GitHub app.
+        repo_owner
+            A GitHub organization.
+        repo_name
+            A GitHub repo.
+        ref
+            A GitHub ref.
+        """
+        client = await factory.create_installation_client(
+            installation_id=installation_id,
+        )
+        return cls(
+            client=client, repo_name=repo_name, repo_owner=repo_owner, ref=ref
+        )
+
+    async def get_changed_files(self) -> list[Path]:
+        """Get a list of files whose contents have changed.
+
+        Returns
+        -------
+        list[Path]
+            List of paths relative to the repo root.
+        """
+        path = f"{self._api_path}/commits/{self.ref}"
+        files = [
+            _ChangedFileResponse.model_validate(info)
+            async for info in self.client.getiter(path, iterable_key="files")
+        ]
+        return [
+            file.filename
+            for file in files
+            if file.status
+            in (
+                _FileStatus.modified,
+                _FileStatus.changed,
+                _FileStatus.added,
+            )
+        ]
+
+    async def get_file_content(self, path: Path) -> str:
+        """Get the contents of a file in a GitHub repo.
+
+        Raises ``GitHubFileNotFoundError`` if the file doesn't exist.
+
+        Parameters
+        ----------
+        path
+            Path of the file relative to the repo root.
+
+        Returns
+        -------
+        str
+            The contents of the file
+        """
+        qs = urlencode({"ref": self.ref})
+        api_path = f"{self._api_path}/contents/{path}?{qs}"
+        try:
+            res = await self.client.getitem(
+                api_path, url_vars={"ref": self.ref}
+            )
+        except HTTPException as exc:
+            if exc.status_code == 404:
+                raise GitHubFileNotFoundError from None
+        file = _FileContentsResponse.model_validate(res)
+        return file.content
+
+    async def create_check_run(
+        self,
+        name: str,
+        summary: str,
+        details: str | None = None,
+    ) -> "CheckRun":
```

Comment:
If you add `from __future__ import annotations` to the start of every file (other than ones that define FastAPI handlers or dependencies), you can stop thinking about definition ordering for type annotations and stop having to ever quote a type annotation.